### PR TITLE
[compiled autograd] log placeholder origin in verbose

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136741
* __->__ #135663

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @rec

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torch/csrc/dynamo/compiled_autograd.h [hunk 1]
```diff
@@ -175,35 +198,63 @@ struct LiftedIValueArg {
...
   size_t next = 0;
+  // NodeCall id of each arg, only when verbose logging is enabled
+  std::vector<uint32_t> args_origins;
+
+ private:
+  const std::optional<size_t>& active_node_call_idx;
 };
 
 struct AutogradCompilerCall {
+  AutogradCompilerCall()
```
Comment (isuruf) [inline torch/csrc/dynamo/compiled_autograd.h:228]: Should this have been something like
`AutogradCompilerCall(const std::optional<size_t>& active_node_call_idx)`

Comment (xmfan) [reply torch/csrc/dynamo/compiled_autograd.h:228]: No, there should be 1 AutogradCompilerCall instance for the entire autograd graph. Is this causing an error?

Comment (isuruf) [reply torch/csrc/dynamo/compiled_autograd.h:228]: I'm seeing the following warning (not an error)
```
In file included from /home/isuruf/git/pytorch/torch/csrc/autograd/custom_function.h:10,
                 from /home/isuruf/git/pytorch/torch/csrc/autograd/python_function.h:6,
                 from /home/isuruf/git/pytorch/torch/csrc/autograd/python_autograd.h:14,
                 from /home/isuruf/git/pytorch/torch/csrc/THP.h:28,
                 from /home/isuruf/git/pytorch/torch/csrc/jit/python/pybind.h:9,
                 from /home/isuruf/git/pytorch/torch/csrc/lazy/python/init.cpp:5:
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h: In constructor 'torch::dynamo::autograd::AutogradCompilerCall::AutogradCompilerCall()':
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h:229:21: warning: member 'torch::dynamo::autograd::AutogradCompilerCall::active_node_call_idx' is used uninitialized [-Wuninitialized]
  229 |       : tensor_args(active_node_call_idx),
      |                     ^~~~~~~~~~~~~~~~~~~~
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h:230:28: warning: member 'torch::dynamo::autograd::AutogradCompilerCall::active_node_call_idx' is used uninitialized [-Wuninitialized]
  230 |         lifted_ivalue_args(active_node_call_idx) {}
      |                            ^~~~~~~~~~~~~~~~~~~~
```

Comment (xmfan) [reply torch/csrc/dynamo/compiled_autograd.h:228]: strange, i can't repro -Wuninitialized, but this should fix it: https://github.com/pytorch/pytorch/pull/137539

**Pull request closed without merging**